changelogs/fragments/2987-archive-stage-idempotency-fix.yml

```
@@ -0,0 +1,3 @@
+---
+minor_changes:
+  - archive - refactoring prior to fix for idempotency checks (https://github.com/ansible-collections/community.general/pull/2987).
```

How about mentioning that the fix has to wait:
```suggestion
  - archive - refactoring prior to fix for idempotency checks. The fix will be a breaking change and only appear in community.general 4.0.0 (https://github.com/ansible-collections/community.general/pull/2987).
```
(in case someone sees the message and hopes it shows up in the next minor release :) )